eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -237,177 +404,591 @@ object Transactions {
+                       htlc: UpdateAddHtlc,
+                       feerate: FeeratePerKw,
+                       commitmentFormat: CommitmentFormat): Either[TxGenerationSkipped, ClaimHtlcTimeoutTx] = {
+      findInput(commitTx, outputs, htlc) match {
+        case Some(input) =>
+          val amount = input.txOut.amount - weight2fee(feerate, commitmentFormat.claimHtlcTimeoutWeight)
+          val tx = Transaction(
+            version = 2,
+            txIn = TxIn(input.outPoint, ByteVector.empty, getHtlcTxInputSequence(commitmentFormat)) :: Nil,
+            txOut = TxOut(amount, localFinalScriptPubKey) :: Nil,
+            lockTime = htlc.cltvExpiry.toLong
+          )
+          val unsignedTx = ClaimHtlcTimeoutTx(input, tx, htlc.paymentHash, htlc.id, htlc.cltvExpiry)
+          skipTxIfBelowDust(unsignedTx, dustLimit, () => unsignedTx.sign(commitKeys, commitmentFormat))
+        case None => Left(OutputNotFound)
+      }
+    }
+  }
+
+  /** This transaction claims our anchor output in either the local or remote commitment, to CPFP and get it confirmed. */
   case class ClaimAnchorOutputTx(input: InputInfo, tx: Transaction, confirmationTarget: ConfirmationTarget) extends ReplaceableTransactionWithInputInfo {
     override val desc: String = "local-anchor"
 
-    def addSigs(localSig: ByteVector64): ClaimAnchorOutputTx = input match {
-      case InputInfo.SegwitInput(_, _, redeemScript) =>
-        val witness = witnessAnchor(localSig, redeemScript)
-        copy(tx = tx.updateWitness(0, witness))
-      case _: InputInfo.TaprootInput => ???
+    def sign(fundingKey: PrivateKey, commitKeys: LocalCommitmentKeys, commitmentFormat: CommitmentFormat, extraUtxos: Map[OutPoint, TxOut]): ClaimAnchorOutputTx = {
+      commitmentFormat match {
+        case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat => sign(fundingKey, commitmentFormat, extraUtxos)
+      }
+    }
+
+    def sign(fundingKey: PrivateKey, commitKeys: RemoteCommitmentKeys, commitmentFormat: CommitmentFormat, extraUtxos: Map[OutPoint, TxOut]): ClaimAnchorOutputTx = {
+      commitmentFormat match {
+        case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat => sign(fundingKey, commitmentFormat, extraUtxos)
+      }
+    }
+
+    private def sign(anchorKey: PrivateKey, commitmentFormat: CommitmentFormat, extraUtxos: Map[OutPoint, TxOut]): ClaimAnchorOutputTx = {
+      val witness = commitmentFormat match {
+        case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat =>
+          val redeemScript = Script.write(anchor(anchorKey.publicKey))
+          val sig = sign(anchorKey, sighash(TxOwner.Local, commitmentFormat), RedeemInfo.P2wsh(redeemScript), extraUtxos)
+          witnessAnchor(sig, redeemScript)
+      }
+      copy(tx = tx.updateWitness(inputIndex, witness))
+    }
+  }
+
+  object ClaimAnchorOutputTx {
+    def redeemInfo(fundingKey: PrivateKey, commitKeys: CommitmentPublicKeys, commitmentFormat: CommitmentFormat): RedeemInfo = {
```

Comment:
We need 2 `redeemInfo` methods, same as `sign`: one that takes a `LocalCommitmentKeys` parameter, and the other that takes a `RemoteCommitmentKeys` parameter.

Reply:
Yes it's probably simpler modify this later.